clarity/src/vm/ast/mod.rs

```diff
@@ -80,6 +81,88 @@ pub fn build_ast<T: CostTracker>(
     Ok(contract_ast)
 }
 
+pub fn build_ast_with_diagnostics<T: CostTracker>(
```

Comment:
Yes, there should be. Good point.

Reply:
Hmmm... I'm not loving the options that I've come up with for this. Did you have anything in mind?

Reply:
Why not just create an `inner_build_ast()` function that took a `with_diagnostics: bool`, and if it was `true`, it would gather the diagnostics?  It would return a `Vec<Diagnostic>`, but if `with_diagnostics` is `false`, it would be empty.  Then, `build_ast()` and `build_ast_with_diagnostics()` just wrap this method.

Reply:
I just thought it got ugly with those other checks in there, but if you both agree that the benefits of a merged source outweigh the ugliness, I'm convinced. I'll make this change.

Reply:
abf68cc05